### PR TITLE
Various sys info widget fixes

### PR DESCRIPTION
**Pull request by phil-davis**
Signed-off-by: Phil Davis <phil@jankaritech.com>

Some are needed just for the backport.
Some are bugs in the master code.

### CODE REVIEW

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ function get_sysinfo_item_html($itemkey) {
 				$args[] = php_uname("m");
 				$args[] = gettext('built on') . ' '. file_get_contents("/etc/version.buildtime");
 				$args[] = (!$g['hideuname'] ? '<br /><span title="' . php_uname("a") . '">(' . php_uname("s") . ' ' . php_uname("r") . ")</span>" : "");
-				$args[] = (!isset($config['system']['firmware']['disablecheck']) ? "<br /><br /><div id='system-information-widget-updatestatus'>" . gettext("Obtaining update status ") . '<i class="fa fa-cog fa-spin"></i></div>' : "");
+				$args[] = (!isset($config['system']['firmware']['disablecheck']) ? "<br /><br /><div id='updatestatus'>" . gettext("Obtaining update status ") . '<i class="fa fa-cog fa-spin"></i></div>' : "");
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:152]: This just needs to go in the backport

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 2]
```diff
@@ -236,7 +236,8 @@ function get_sysinfo_item_html($itemkey) {
 		</div>
 		<span id="mbufusagemeter">%s</span>&nbsp;<span id="mbuf">(%s)</span>
 END;
-				get_mbuf($mbufstext, $mbufusage);
+				$mbufstext = get_mbuf();
+				$mbufusage = get_mbuf(true);
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:240]: get_mbuf() is different in RELENG_2_3 so this needs to be just in the backport.

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 3]
```diff
@@ -334,6 +335,7 @@ function get_sysinfo_item_html($itemkey) {
 				break;
 
 			case 'ALL_HIDDEN':
+				$title_content = gettext('Empty');
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:338]: Helps to see something in the "item title" for the empty item. Can go in master.

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 4]
```diff
@@ -435,15 +437,14 @@ function get_sysinfo_item_html($itemkey) {
 		$catsort = $catdata['catsort'];
 		$data[$catsort]['cattitle'] = $catdata['cattitle']; // add key for category and get category title
 		$data[$catsort]['itemstoshow'][$itemsort] = $itemdata;  // create a key for item and add static item data
-		$data[$catsort]['itemstoshow'][$itemsort]['item_html'] = '<tr>' . $item_html . '<tr>';
+		$data[$catsort]['itemstoshow'][$itemsort]['item_html'] = '<tr>' . $item_html . '</tr>';
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:440]: Needs to be fixed in master PR also.

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 5]
```diff
@@ -435,15 +437,14 @@ function get_sysinfo_item_html($itemkey) {
...
 }
 if (count($data) == 0) {
 	// adds a single item containing the special key for the "no sysinfo selected" item if nothing else will display
-	$data[0] = array(
-		'cattitle' => '',
-		'itemstoshow' => array('itemsort' => '', 'item_html' => '<tr>' . get_sysinfo_item_html('ALL_HIDDEN') . '<tr>')
-		);
+	$data[0]['cattitle'] = '';
+	$data[0]['itemstoshow'][0]['itemtitle'] = '';
+	$data[0]['itemstoshow'][0]['item_html'] = '<tr>' . get_sysinfo_item_html('ALL_HIDDEN') . '</tr>';
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:447]: This creates the proper data structure to match what is expected below. The other array method above did not work with the code below. This needs to be fixed in the master PR also.

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 6]
```diff
@@ -456,8 +457,10 @@ function get_sysinfo_item_html($itemkey) {
 	//sort categories
 	ksort($data);
 	foreach ($data as $cat => $cat_data) {
-		// display title for this category
-		echo "<tr><th><strong>{$cat_data['cattitle']}</strong></th>\n<td>&nbsp;</td></tr>";
+		if (strlen($cat_data['cattitle']) > 0) {
+			// display title for this category
+			echo "<tr><th><strong>{$cat_data['cattitle']}</strong></th>\n<td>&nbsp;</td></tr>";
+		}
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:463]: Only display the cattitle if it has something in it. Handles the "no items to display" case when there is no cattitle. Needs to go back into master PR.

File: src/usr/local/www/widgets/widgets/system_information.widget.php [hunk 7]
```diff
@@ -474,7 +477,6 @@ function get_sysinfo_item_html($itemkey) {
 </div><div id="widget-<?=$widgetname?>_panel-footer" class="panel-footer collapse">
 
 <form action="/widgets/widgets/system_information.widget.php" method="post" class="form-horizontal">
-<form action="/widgets/widgets/system_information.widget.php" method="post" class="form-horizontal">
```
Comment (phil-davis) [inline src/usr/local/www/widgets/widgets/system_information.widget.php:477]: This line was duplicated. Check the master PR in case it is duplicated there.

**Comment by phil-davis** [comment]
@stilez the challenge is to sort out which things are fixes to put back into the master PR. I kept noticing stuff and then realizing that actually it did not work properly in the master PR for 2.4

File: src/usr/local/www/widgets/widgets/system_information.widget.php (hunk 7 above)
Comment (stilez) [reply src/usr/local/www/widgets/widgets/system_information.widget.php:477]: no, there was a custom title (new widget system) taken out, I probably replaced it not noticing the original line was just above it

**Pull request merged**